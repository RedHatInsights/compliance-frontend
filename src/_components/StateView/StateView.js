import React from 'react';

export const StateView = ({ stateValues, children }) => (
  <>
    {children
      .flatMap((c) => c)
      .filter((child) => stateValues[child.props.stateKey])}
  </>
);

StateView.propTypes = props;

export default StateView;
