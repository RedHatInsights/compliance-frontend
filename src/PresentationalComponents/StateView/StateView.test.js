import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import { StateView, StateViewPart, StateViewWithError } from './StateView';

describe('StatView', () => {
  it('expect to render loading', () => {
    render(
      <StateView
        stateValues={{
          loading: true,
          data: undefined,
          error: undefined,
        }}
      >
        <StateViewPart stateKey="loading">LOADING</StateViewPart>
        <StateViewPart stateKey="data">DATA</StateViewPart>
        <StateViewPart stateKey="error">ERROR</StateViewPart>
      </StateView>
    );

    expect(screen.getByText('LOADING')).toBeInTheDocument();
    expect(screen.queryByText('DATA')).not.toBeInTheDocument();
    expect(screen.queryByText('ERROR')).not.toBeInTheDocument();
  });

  it('expect to render error', () => {
    render(
      <StateView
        stateValues={{
          loading: undefined,
          error: { error: 'ERROR' },
          data: undefined,
        }}
      >
        <StateViewPart stateKey="loading">LOADING</StateViewPart>
        <StateViewPart stateKey="data">DATA</StateViewPart>
        <StateViewPart stateKey="error">ERROR</StateViewPart>
      </StateView>
    );

    expect(screen.getByText('ERROR')).toBeInTheDocument();
    expect(screen.queryByText('LOADING')).not.toBeInTheDocument();
    expect(screen.queryByText('DATA')).not.toBeInTheDocument();
  });

  it('expect to render data', () => {
    render(
      <StateView
        stateValues={{
          loading: undefined,
          error: undefined,
          data: { data: 'data' },
        }}
      >
        <StateViewPart stateKey="loading">LOADING</StateViewPart>
        <StateViewPart stateKey="data">DATA</StateViewPart>
        <StateViewPart stateKey="error">ERROR</StateViewPart>
      </StateView>
    );

    expect(screen.getByText('DATA')).toBeInTheDocument();
    expect(screen.queryByText('LOADING')).not.toBeInTheDocument();
    expect(screen.queryByText('ERROR')).not.toBeInTheDocument();
  });
});

describe('StatViewWithError', () => {
  it('expect to render error page', () => {
    render(
      <StateViewWithError
        stateValues={{
          loading: undefined,
          error: { error: 'ERROR' },
          data: undefined,
        }}
      >
        <StateViewPart stateKey="loading">LOADING</StateViewPart>
        <StateViewPart stateKey="data">DATA</StateViewPart>
      </StateViewWithError>
    );

    expect(screen.getByText('Something went wrong')).toBeInTheDocument();
    expect(screen.queryByText('LOADING')).not.toBeInTheDocument();
    expect(screen.queryByText('DATA')).not.toBeInTheDocument();
  });
});
