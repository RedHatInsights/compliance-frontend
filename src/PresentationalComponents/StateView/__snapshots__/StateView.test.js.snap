// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatView expect to render data 1`] = `
<StateViewPart
  stateKey="data"
>
  DATA
</StateViewPart>
`;

exports[`StatView expect to render error 1`] = `
<StateViewPart
  stateKey="error"
>
  ERROR
</StateViewPart>
`;

exports[`StatView expect to render loading 1`] = `
<StateViewPart
  stateKey="loading"
>
  LOADING
</StateViewPart>
`;

exports[`StatViewWithError expect to render error page 1`] = `
<StateView
  stateValues={
    Object {
      "data": undefined,
      "error": Object {
        "error": "ERROR",
      },
      "loading": undefined,
    }
  }
>
  <StateViewPart
    key="error-state"
    stateKey="error"
  >
    <ErrorPage
      error={
        Object {
          "error": "ERROR",
        }
      }
    />
  </StateViewPart>
  <StateViewPart
    stateKey="loading"
  >
    LOADING
  </StateViewPart>
  <StateViewPart
    stateKey="data"
  >
    DATA
  </StateViewPart>
</StateView>
`;
